### PR TITLE
Support extra hygiene for signature(::Methood)

### PR DESCRIPTION
**Pull request by oxinabox**
This is adding the extra-hygiene feature from #18 
It is the thing to work around https://github.com/JuliaLang/julia/issues/39876

### CODE REVIEW

File: src/method.jl [hunk 1]
```diff
@@ -27,13 +27,23 @@ with [`CodeTracking.definition`](https://github.com/timholy/CodeTracking.jl).
 The dictionary of components returned by `signature` match those returned by
 [`splitdef`](@ref) and include all that are required by [`combinedef`](@ref), except for
 the `:body` component.
+
+# keywords
+
+ - `extra_hygiene=false`: if set to `true` this forces name-hygine on the `TypeVar`s in
```
Comment (mzgubic) [inline src/method.jl]:
```suggestion
 - `extra_hygiene=false`: if set to `true` this forces name-hygiene on the `TypeVar`s in 
```

**Review by mzgubic — COMMENTED** [review]
Wait, was this not added in https://github.com/invenia/ExprTools.jl/pull/18? It sure looks like it was in the PR

**Review by mzgubic — APPROVED** [review]
also needs version bump

File: src/method.jl (hunk 1 above)
Comment (oxinabox) [reply src/method.jl]: I may have copied that from a earlier revision

**Pull request merged**